src/codec_aom.c

```diff
@@ -830,6 +879,7 @@ static avifResult aomCodecEncodeImage(avifCodec * codec,
     }
     aom_codec_err_t encodeErr = aom_codec_encode(&codec->internal->encoder, &aomImage, 0, 1, encodeFlags);
     avifFree(monoUVPlane);
+    aom_img_free(&aomImage);
```

Comment:
Is this safe to call here in all cases? This seems like it'll perform frees on memory it doesn't own?

Reply:
This turns out to be safe, because `aom_img_free` is defined as follows:

```
void aom_img_free(aom_image_t *img) {
  if (img) {
    aom_img_remove_metadata(img);
    if (img->img_data && img->img_data_owner) aom_free(img->img_data);

    if (img->self_allocd) free(img);
  }
}
```

I added a test for `aomImageAllocated` to make this clearer.